<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/junsu/etasl/ws/etasl-py/src/etasl_py_examples/robots/ur10_robot.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur10">
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
    <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
  </gazebo>
  <!--
  Author: Kelsey Hawkins
  Contributers: Jimmy Da Silva, Ajit Krisshna N L, Muhammad Asif Rana
-->
  <!-- measured from model -->
  <link name="world"/>
  
  
  
  <!-- robot 1 -->
  <link name="base_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint_1" type="revolute">
    <parent link="base_link_1"/>
    <child link="shoulder_link_1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1273"/>
    <axis xyz="0 0 1"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="7.778"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0314743125769" ixy="0.0" ixz="0.0" iyy="0.0314743125769" iyz="0.0" izz="0.021875625"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint_1" type="revolute">
    <parent link="shoulder_link_1"/>
    <child link="upper_arm_link_1"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.220941 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="12.93"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.306"/>
      <inertia ixx="0.421753803798" ixy="0.0" ixz="0.0" iyy="0.421753803798" iyz="0.0" izz="0.036365625"/>
    </inertial>
  </link>
  <joint name="elbow_joint_1" type="revolute">
    <parent link="upper_arm_link_1"/>
    <child link="forearm_link_1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1719 0.612"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.87"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28615"/>
      <inertia ixx="0.111069694097" ixy="0.0" ixz="0.0" iyy="0.111069694097" iyz="0.0" izz="0.010884375"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint_1" type="revolute">
    <parent link="forearm_link_1"/>
    <child link="wrist_1_link_1"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.5723"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint_1" type="revolute">
    <parent link="wrist_1_link_1"/>
    <child link="wrist_2_link_1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.1149 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint_1" type="revolute">
    <parent link="wrist_2_link_1"/>
    <child link="wrist_3_link_1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1157"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link_1">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.202"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000526462289415" ixy="0.0" ixz="0.0" iyy="0.000526462289415" iyz="0.0" izz="0.000568125"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint_1" type="fixed">
    <parent link="wrist_3_link_1"/>
    <child link="ee_link_1"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.0922 0.0"/>
  </joint>
  <link name="ee_link_1">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="shoulder_pan_trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor_1">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor_1">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor_1">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor_1">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor_1">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor_1">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="shoulder_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="upper_arm_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="forearm_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_1_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_3_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_2_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="ee_link_1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base_1"/>
  <joint name="base_link-base_fixed_joint_1" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="base_link_1"/>
    <child link="base_1"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0_1"/>
  <joint name="wrist_3_link-tool0_fixed_joint_1" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.0922 0"/>
    <parent link="wrist_3_link_1"/>
    <child link="tool0_1"/>
  </joint>
  <joint name="world_joint_1" type="fixed">
    <parent link="world"/>
    <child link="base_link_1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.5 0.0"/>
  </joint>
  
  
  
  
  
  
  <!-- robot 2 -->
  <link name="base_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint_2" type="revolute">
    <parent link="base_link_2"/>
    <child link="shoulder_link_2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1273"/>
    <axis xyz="0 0 1"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="7.778"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0314743125769" ixy="0.0" ixz="0.0" iyy="0.0314743125769" iyz="0.0" izz="0.021875625"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint_2" type="revolute">
    <parent link="shoulder_link_2"/>
    <child link="upper_arm_link_2"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.220941 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="12.93"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.306"/>
      <inertia ixx="0.421753803798" ixy="0.0" ixz="0.0" iyy="0.421753803798" iyz="0.0" izz="0.036365625"/>
    </inertial>
  </link>
  <joint name="elbow_joint_2" type="revolute">
    <parent link="upper_arm_link_2"/>
    <child link="forearm_link_2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1719 0.612"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.87"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28615"/>
      <inertia ixx="0.111069694097" ixy="0.0" ixz="0.0" iyy="0.111069694097" iyz="0.0" izz="0.010884375"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint_2" type="revolute">
    <parent link="forearm_link_2"/>
    <child link="wrist_1_link_2"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.5723"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint_2" type="revolute">
    <parent link="wrist_1_link_2"/>
    <child link="wrist_2_link_2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.1149 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint_2" type="revolute">
    <parent link="wrist_2_link_2"/>
    <child link="wrist_3_link_2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1157"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link_2">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.202"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000526462289415" ixy="0.0" ixz="0.0" iyy="0.000526462289415" iyz="0.0" izz="0.000568125"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint_2" type="fixed">
    <parent link="wrist_3_link_2"/>
    <child link="ee_link_2"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.0922 0.0"/>
  </joint>
  <link name="ee_link_2">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="shoulder_pan_trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor_2">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor_2">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor_2">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor_2">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor_2">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor_2">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="shoulder_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="upper_arm_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="forearm_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_1_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_3_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_2_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="ee_link_2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base_2"/>
  <joint name="base_link-base_fixed_joint_2" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="base_link_2"/>
    <child link="base_2"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0_2"/>
  <joint name="wrist_3_link-tool0_fixed_joint_2" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.0922 0"/>
    <parent link="wrist_3_link_2"/>
    <child link="tool0_2"/>
  </joint>
  <joint name="world_joint_2" type="fixed">
    <parent link="world"/>
    <child link="base_link_2"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.5 0.0"/>
  </joint>
  
  
</robot>
